### PR TITLE
Kids First DRS Signed URLs [BT-172] [BT-218]

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

File: martha/martha_v3.js [hunk 1]
```diff
@@ -538,7 +550,9 @@ async function retrieveFromServers(params) {
     }
 
     let bondSA;
-    if (bondProvider && overlapFields(requestedFields, MARTHA_V3_BOND_SA_FIELDS)) {
+    if (bondProvider &&
+        accessMethodType !== ACCESS_METHOD_TYPE_S3 && // do not get Google SA if this is S3
```
Comment (kshakir) [inline martha/martha_v3.js]: This should be something like (comments need re-english)
```javascript
bondProvider && // Only retrieve the SA if the DRS provider uses a Bond provider for auth
[ACCESS_METHOD_TYPE_NONE, ACCESS_METHOD_TYPE_GCS].includes(accessMethodType) && // Only retrieve the SA for projects that implicitly or explicitly use GCS for accessing the data
```

File: test/martha/martha_v3.test.js [hunk 2]
```diff
@@ -857,24 +864,33 @@ test.serial('martha_v3 parses Kids First response correctly', async (t) => {
...
+        `https://${config.HOST_KIDS_FIRST_STAGING}/ga4gh/drs/v1/objects/ed6be7ab-068e-46c8-824a-f39cfbb885cc`,
     );
     t.falsy(getJsonFromApiStub.getCall(0).args[1]); // no auth passed
-    const requestedBondUrl = getJsonFromApiStub.getCall(1).args[0];
-    const matches = requestedBondUrl.match(bondRegEx);
-    t.truthy(matches, 'Bond URL called does not match Bond URL regular expression');
-    const expectedProvider = 'dcf-fence';
-    const actualProvider = matches[2];
-    t.is(actualProvider, expectedProvider);
+    const requestedBondAccessToken = getJsonFromApiStub.getCall(1).args[0];
```
Comment (kshakir) [inline test/martha/martha_v3.test.js]: While making other edits, please add a newline between checks for `getCall(0)` and `getCall(1)` to match the spacing between `getCall(2)` below.

File: test/martha/martha_v3.test.js [hunk 3]
```diff
@@ -883,19 +899,25 @@ test.serial('martha_v3 parses a Kids First CIB URI response correctly', async (t
...
+    const accessTokenMatches = requestedBondAccessToken.match(bondAccessTokenRegEx);
+    t.truthy(accessTokenMatches, 'Bond URL called does not match Bond URL regular expression');
+    const expectedAccessTokenProvider = 'kids-first';
+    const actualAccessTokenProvider = accessTokenMatches[2];
+    t.is(actualAccessTokenProvider, expectedAccessTokenProvider);
+    t.is(
+        getJsonFromApiStub.getCall(2).args[0],
```
Comment (kshakir) [inline test/martha/martha_v3.test.js]: Minor whitespace consistency again: add a space before and after the checks during `getCall(1)` to separate the section from `getCall(0)` and `getCall(2)`.

**Review by breilly2 — COMMENTED** [review]
Looks good to me so far. The only question I know of is if our suggestion of `kids-first` as the name for the new bond provider will be used. Hopefully we can get a definitive answer to that before we find ourselves blocked by not merging this PR.

**Review by kshakir — APPROVED** [review]

**Comment by mcovarr** [comment]
jenkins retest

**Review by kshakir — APPROVED** [review]
Re-👍

**Review by breilly2 — APPROVED** [review]
👍

**Pull request merged**